desktop-exporter/trace_data.go

```diff
@@ -4,37 +4,37 @@ import (
 	"time"
 )
 
-func (trace *TraceData) GetTraceSummary() (TraceSummary, error) {
-	duration, err := trace.getTraceDuration()
-	if err != nil {
-		return TraceSummary{}, err
+func (trace *TraceData) GetTraceSummary() (TraceSummary) {
+	rootSpan, err := trace.getRootSpan()
+	
+	if err == ErrMissingRootSpan {
+		return TraceSummary{
+			HasRootSpan:     false,
+			RootServiceName: "",
+			RootName:        "",
+			RootStartTime:   time.Time{},
+			RootEndTime:	 time.Time{},
+			SpanCount:       uint32(len(trace.Spans)),
+			TraceID:         trace.TraceID,
+		}
 	}
-
+	
 	return TraceSummary{
-		TraceID:    trace.TraceID,
-		SpanCount:  uint32(len(trace.Spans)),
-		DurationMS: duration.Milliseconds(),
-	}, nil
-
-}
-
-func (trace *TraceData) getTraceDuration() (time.Duration, error) {
-	if len(trace.Spans) < 1 {
-		return 0, ErrEmptySpansSlice
+		HasRootSpan:     true,
+		RootServiceName: rootSpan.Resource.Attributes["service.name"].(string),
```

Comment:
What happens when this isn't present? What happens if this is somehow set to a bool or float? 🤔

Reply:
Consider factoring this out into a method on `SpanData` https://github.com/CtrlSpice/otel-desktop-viewer/blob/main/desktop-exporter/types.go#L39

which will make easy to test in isolation and simplifies this line to:

```go
RootServiceName: rootSpan.ServiceName()
```

Reply:
Hey, so it is my understanding that if it is somehow set to a float or a bool, or isn't present, there is an issue in the SDK because it's supposed to fall back to `unknown_service :` + `process.executable.name` or failing that, just `unknown_service`. I've factored it out though, and set it to warn about missing or invalid service names.

Here's the thing I read: https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/#service
I decided not to fall back on `unknown_service` in the app, since that could mask a problem for someone if they are using my thing to check if they've instrumented their code correctly? Does that sound right? I just output an empty string...

What do you think?